### ISSUE TITLE
DASH+DRM+CMAF content can't be played

### ISSUE THREAD

**Issue by CMMB93** [google/ExoPlayer#10266]
### ExoPlayer Version

Older (unsupported)

### Devices that reproduce the issue

Samsung Galaxy S9  running Android X
generic_x86_arm, sdk_gphone_x86, Google, 30

### Reproducible in the demo app?

Yes

### Reproduction steps

We tried to play the video after implementing our Widevine license URL, proxy URL, and Portal ID.
We selected a CMAF film, much more precisely a DASH-MPD manifest.

### Expected result

The video to be played.

### Actual result

Black screen, error message
Only the duration is found. 

D/MainActivity: ExoPlayer 2 pressed
W/exoplayersampl: Suspending all threads took: 18.953ms
I/TetheringManager: registerTetheringEventCallback:com.exoplayersample
W/exoplayersampl: Suspending all threads took: 5.145ms
W/exoplayersampl: Accessing hidden method Landroid/graphics/FontFamily;-><init>()V (greylist-max-q, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/graphics/FontFamily;->addFontFromAssetManager(Landroid/content/res/AssetManager;Ljava/lang/String;IZIII[Landroid/graphics/fonts/FontVariationAxis;)Z (greylist-max-q, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/graphics/FontFamily;->addFontFromBuffer(Ljava/nio/ByteBuffer;I[Landroid/graphics/fonts/FontVariationAxis;II)Z (greylist-max-q, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/graphics/FontFamily;->freeze()Z (greylist-max-q, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/graphics/FontFamily;->abortCreation()V (greylist-max-q, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/graphics/Typeface;->createFromFamiliesWithDefault([Landroid/graphics/FontFamily;Ljava/lang/String;II)Landroid/graphics/Typeface; (greylist, reflection, allowed)
W/exoplayersampl: Accessing hidden method Landroid/media/AudioTrack;->getLatency()I (greylist, reflection, allowed)
I/ExoPlayerImpl: Init 612ac15 [ExoPlayerLib/2.11.1] [generic_x86_arm, sdk_gphone_x86, Google, 30]
D/ExoPlayer2EventListener: onPlayerStateChanged() called with: playWhenReady = [true], playbackState = [1]
I/DrmUtils: found android.hardware.drm@1.0::IDrmFactory clearkey
I/DrmUtils: found android.hardware.drm@1.0::IDrmFactory default
I/DrmUtils: found android.hardware.drm@1.0::IDrmFactory widevine
I/DrmUtils: found android.hardware.drm@1.1::IDrmFactory clearkey
I/DrmUtils: found android.hardware.drm@1.1::IDrmFactory widevine
I/DrmUtils: found android.hardware.drm@1.2::IDrmFactory clearkey
I/DrmUtils: found android.hardware.drm@1.2::IDrmFactory widevine
I/DrmUtils: found android.hardware.drm@1.3::IDrmFactory clearkey
I/DrmUtils: found android.hardware.drm@1.3::IDrmFactory widevine
D/ExoPlayer2EventListener: onPlayerStateChanged() called with: playWhenReady = [true], playbackState = [2]
I/Choreographer: Skipped 46 frames!  The application may be doing too much work on its main thread.
W/exoplayersampl: Suspending all threads took: 6.696ms
I/OpenGLRenderer: Davey! duration=907ms; Flags=1, IntendedVsync=9536719780488, Vsync=9537486447124, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=9537497413120, AnimationStart=9537497482120, PerformTraversalsStart=9537498442120, DrawStart=9537567241120, SyncQueued=9537591563120, SyncStart=9537594906120, IssueDrawCommandsStart=9537595503120, SwapBuffers=9537626787120, FrameCompleted=9537630496120, DequeueBufferDuration=1266000, QueueBufferDuration=896000, GpuCompleted=72904454231491230, 
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [null], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@5157332], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@de77883]
W/exoplayersampl: Suspending all threads took: 7.038ms
W/exoplayersampl: Suspending all threads took: 6.181ms
D/ExoPlayer2EventListener: onLoadCompleted() called with: windowIndex = [0], mediaPeriodId = [null], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@4ad0ce2], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@678873]
D/ExoPlayer2EventListener: onTimelineChanged() called with: timeline = [com.google.android.exoplayer2.source.dash.DashMediaSource$DashTimeline@39a5330], reason = [0]
D/ExoPlayer2EventListener: onMediaPeriodCreated() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e]
D/ExoPlayer2EventListener: onLoadingChanged() called with: isLoading = [true]
I/VideoCapabilities: Unsupported profile 4 for video/mp4v-es
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@609725c], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@2bf563a]
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@29992eb], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@2e95848]
D/ExoPlayer2EventListener: onTracksChanged() called
D/ExoPlayer2EventListener: onReadingStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e]
D/ExoPlayer2EventListener: onLoadCompleted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@9f67292], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@3c37463]
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@e0525de], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@16ef3bf]
D/ExoPlayer2EventListener: onDownstreamFormatChanged() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@f3c2a8c]
D/ExoPlayer2EventListener: onDrmSessionAcquired() called
D/CryptoHal: found drm@1.0 ICryptoFactory clearkey
D/CryptoHal: found drm@1.0 ICryptoFactory default
D/CryptoHal: found drm@1.0 ICryptoFactory widevine
D/CryptoHal: found drm@1.1 ICryptoFactory clearkey
D/CryptoHal: found drm@1.1 ICryptoFactory widevine
I/OMXClient: IOmx service obtained
D/SurfaceUtils: connecting to surface 0xe940e2c8, reason connectToSurface
I/MediaCodec: [OMX.android.goldfish.h264.decoder] setting surface generation to 10702849
D/SurfaceUtils: disconnecting from surface 0xe940e2c8, reason connectToSurface(reconnect)
D/SurfaceUtils: connecting to surface 0xe940e2c8, reason connectToSurface(reconnect)
E/ACodec: [OMX.android.goldfish.h264.decoder] setPortMode on output to DynamicANWBuffer failed w/ err -1010
I/ACodec: codec does not support config priority (err -1010)
D/SurfaceUtils: disconnecting from surface 0xe940e2c8, reason setNativeWindowSizeFormatAndUsage
D/SurfaceUtils: connecting to surface 0xe940e2c8, reason setNativeWindowSizeFormatAndUsage
D/SurfaceUtils: set up nativeWindow 0xe940e2c8 for 1280x720, color 0x13, rotation 0, usage 0x1002900
D/ExoPlayer2EventListener: onLoadCompleted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@9b9c1ea], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@2290cdb]
D/ExoPlayer2EventListener: onLoadCompleted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@9c76378], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@1041a51]
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@a6a12b6], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@fa19bb7]
D/ExoPlayer2EventListener: onLoadStarted() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@bcb3f24], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@c9f0a8d]
D/ExoPlayer2EventListener: onDownstreamFormatChanged() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@afca442]
D/CryptoHal: found drm@1.0 ICryptoFactory clearkey
D/CryptoHal: found drm@1.0 ICryptoFactory default
D/CryptoHal: found drm@1.0 ICryptoFactory widevine
D/CryptoHal: found drm@1.1 ICryptoFactory clearkey
D/CryptoHal: found drm@1.1 ICryptoFactory widevine
W/exoplayersampl: Suspending all threads took: 6.133ms
D/CCodec: allocate(c2.android.aac.decoder)
I/Codec2Client: Available Codec2 services: "software"
I/CCodec: Created component [c2.android.aac.decoder]
D/CCodecConfig: read media type: audio/mp4a-latm
D/ReflectedParamUpdater: extent() != 1 for single value type: algo.buffers.max-count.values
D/ReflectedParamUpdater: extent() != 1 for single value type: output.subscribed-indices.values
D/ReflectedParamUpdater: extent() != 1 for single value type: input.buffers.allocator-ids.values
D/ReflectedParamUpdater: extent() != 1 for single value type: output.buffers.allocator-ids.values
D/ReflectedParamUpdater: extent() != 1 for single value type: algo.buffers.allocator-ids.values
D/ReflectedParamUpdater: extent() != 1 for single value type: output.buffers.pool-ids.values
D/ReflectedParamUpdater: extent() != 1 for single value type: algo.buffers.pool-ids.values
I/CCodecConfig: query failed after returning 19 values (BAD_INDEX)
D/CCodecConfig: c2 config diff is Dict {
      c2::u32 coded.aac-packaging.value = 0
      c2::u32 coded.bitrate.value = 64000
      c2::u32 coded.pl.level = 0
      c2::u32 coded.pl.profile = 8192
      c2::i32 coding.drc.album-mode.value = 0
      c2::float coding.drc.attenuation-factor.value = 1
      c2::float coding.drc.boost-factor.value = 1
      c2::i32 coding.drc.compression-mode.value = 3
      c2::i32 coding.drc.effect-type.value = 3
      c2::float coding.drc.encoded-level.value = 0.25
      c2::float coding.drc.reference-level.value = -16
      c2::u32 input.buffers.max-size.value = 8192
      c2::u32 input.delay.value = 0
      string input.media-type.value = "audio/mp4a-latm"
      c2::u32 output.delay.value = 2
      c2::float output.drc.output-loudness.value = 0.25
      string output.media-type.value = "audio/raw"
      c2::u32 raw.channel-count.value = 1
      c2::u32 raw.max-channel-count.value = 8
      c2::u32 raw.sample-rate.value = 44100
    }
D/CCodec: [c2.android.aac.decoder] buffers are bound to CCodec for this session
D/CCodecConfig: no c2 equivalents for flags
D/CCodecConfig: c2 config diff is   c2::u32 raw.channel-count.value = 2
      c2::u32 raw.sample-rate.value = 48000
W/Codec2Client: query -- param skipped: index = 1107298332.
D/CCodec: setup formats input: AMessage(what = 0x00000000) = {
      int32_t aac-drc-album-mode = 0
      int32_t aac-drc-boost-level = 127
      int32_t aac-drc-cut-level = 127
      int32_t aac-drc-effect-type = 3
      int32_t aac-drc-heavy-compression = 3
      int32_t aac-encoded-target-level = -1
      int32_t aac-max-output-channel_count = 8
      int32_t aac-target-ref-level = 64
      int32_t channel-count = 2
      int32_t level = 0
      int32_t max-input-size = 8192
      string mime = "audio/mp4a-latm"
      int32_t profile = 2
      int32_t sample-rate = 48000
    } and output: AMessage(what = 0x00000000) = {
      int32_t aac-drc-album-mode = 0
      int32_t aac-drc-boost-level = 127
      int32_t aac-drc-cut-level = 127
      int32_t aac-drc-effect-type = 3
      int32_t aac-drc-heavy-compression = 3
      int32_t aac-drc-output-loudness = -1
      int32_t aac-encoded-target-level = -1
      int32_t aac-max-output-channel_count = 8
      int32_t aac-target-ref-level = 64
      int32_t channel-count = 2
      string mime = "audio/raw"
      int32_t sample-rate = 48000
    }
W/Codec2Client: query -- param skipped: index = 1342179345.
W/Codec2Client: query -- param skipped: index = 2415921170.
W/Codec2Client: query -- param skipped: index = 1610614798.
E/FMQ: grantorIdx must be less than 3
E/FMQ: grantorIdx must be less than 3
D/CCodecBufferChannel: [c2.android.aac.decoder#902] Created input block pool with allocatorID 16 => poolID 17 - OK (0)
E/ion: ioctl c0044901 failed with code -1: Inappropriate ioctl for device
I/CCodecBufferChannel: [c2.android.aac.decoder#902] Created output block pool with allocatorID 16 => poolID 32 - OK
D/CCodecBufferChannel: [c2.android.aac.decoder#902] Configured output block pool ids 32 => OK
E/FMQ: grantorIdx must be less than 3
E/ExoPlayer2EventListener: onLoadError() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@4aa3c53], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@60f2990], error = [com.google.android.exoplayer2.ParserException: Length mismatch: 0, 141], wasCanceled = [true]
E/ExoPlayerImplInternal: Source error.
    com.google.android.exoplayer2.ParserException: Length mismatch: 0, 141
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseSaiz(FragmentedMp4Extractor.java:812)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseTraf(FragmentedMp4Extractor.java:736)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseMoof(FragmentedMp4Extractor.java:704)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onMoofContainerAtomRead(FragmentedMp4Extractor.java:552)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onContainerAtomRead(FragmentedMp4Extractor.java:465)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.processAtomEnded(FragmentedMp4Extractor.java:443)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.readAtomPayload(FragmentedMp4Extractor.java:438)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:324)
        at com.google.android.exoplayer2.source.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:135)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:391)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
D/SurfaceUtils: disconnecting from surface 0xe940e2c8, reason disconnectFromSurface
D/CCodecBufferChannel: [c2.android.aac.decoder#902] MediaCodec discarded an unknown buffer
I/chatty: uid=10156(com.exoplayersample) MediaCodec_loop identical 2 lines
D/CCodecBufferChannel: [c2.android.aac.decoder#902] MediaCodec discarded an unknown buffer
W/exoplayersampl: Suspending all threads took: 9.810ms
I/hw-BpHwBinder: onLastStrongRef automatically unlinking death recipients
W/exoplayersampl: Suspending all threads took: 7.562ms
D/ExoPlayer2EventListener: onLoadCanceled() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e], loadEventInfo = [com.google.android.exoplayer2.source.MediaSourceEventListener$LoadEventInfo@c34c28e], mediaLoadData = [com.google.android.exoplayer2.source.MediaSourceEventListener$MediaLoadData@d924aaf]
D/ExoPlayer2EventListener: onDrmSessionReleased() called
D/ExoPlayer2EventListener: onMediaPeriodReleased() called with: windowIndex = [0], mediaPeriodId = [com.google.android.exoplayer2.source.MediaSource$MediaPeriodId@1d01ea6e]
W/exoplayersampl: Suspending all threads took: 10.220ms

E/ExoPlayer2EventListener: onPlayerError() called
    com.google.android.exoplayer2.ExoPlaybackException: com.google.android.exoplayer2.ParserException: Length mismatch: 0, 141
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:394)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:223)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: com.google.android.exoplayer2.ParserException: Length mismatch: 0, 141
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseSaiz(FragmentedMp4Extractor.java:812)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseTraf(FragmentedMp4Extractor.java:736)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseMoof(FragmentedMp4Extractor.java:704)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onMoofContainerAtomRead(FragmentedMp4Extractor.java:552)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onContainerAtomRead(FragmentedMp4Extractor.java:465)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.processAtomEnded(FragmentedMp4Extractor.java:443)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.readAtomPayload(FragmentedMp4Extractor.java:438)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:324)
        at com.google.android.exoplayer2.source.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:135)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:391)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
D/ExoPlayer2EventListener: onLoadingChanged() called with: isLoading = [false]
W/exoplayersampl: Suspending all threads took: 6.178ms
D/ExoPlayer2EventListener: onPlayerStateChanged() called with: playWhenReady = [true], playbackState = [1]
D/BufferPoolAccessor2.0: bufferpool2 0xee7cba18 : 0(0 size) total buffers - 0(0 size) used buffers - 0/4 (recycle/alloc) - 1/2 (fetch/transfer)
D/BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
I/exoplayersampl: Waiting for a blocking GC Instrumentation

### Media

sent by e-mail

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to dev.exoplayer@gmail.com after filing this issue.

**Comment by icbaker** [comment]
We didn't receive the media sent by email. Please ensure you sent it to dev.exoplayer@gmail.com using the subject `Issue #10266`. Thanks!

**Comment by CMMB93** [comment]
Thank you for your e-mail, Mr Baker, I sent it via our e-mail address 
***@***.*** .

I can add you to the Git repository as well if you want, I will need an 
e-mail address to add you as a collaborator.

Thank you,

---

Le 17/05/2022 17:10, Ian Baker a écrit :

> We didn't receive the media sent by email. Please ensure you sent it to 
> ***@***.*** using the subject Issue #10266. Thanks!
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://github.com/google/ExoPlayer/issues/10266#issuecomment-1128992167
[2]

**Comment by icbaker** [comment]
I tried playing the content provided in the demo app at version 2.17.1 using the following JSON snippet:

```
[
  {
    "name": "Issue #10266",
    "samples": [
      {
        "name": "content",
        "uri": <MPD URI>,
        "drm_scheme": "widevine",
        "drm_license_uri": <LICENSE URI>,
        "drm_key_request_properties": {<KEY>: <VALUE> }
      }
    ]
  }
]
```

Playback succeeded.

I then tried the same snippet in the demo app at version 2.11.1 (had to change `"drm_license_uri"` to `"drm_license_url"`)  and playback failed with the same error you're seeing:

```
     Caused by: com.google.android.exoplayer2.ParserException: Length mismatch: 0, 141
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseSaiz(FragmentedMp4Extractor.java:812)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseTraf(FragmentedMp4Extractor.java:736)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.parseMoof(FragmentedMp4Extractor.java:704)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onMoofContainerAtomRead(FragmentedMp4Extractor.java:552)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.onContainerAtomRead(FragmentedMp4Extractor.java:465)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.processAtomEnded(FragmentedMp4Extractor.java:443)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.readAtomPayload(FragmentedMp4Extractor.java:438)
        at com.google.android.exoplayer2.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:324)
        at com.google.android.exoplayer2.source.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:135)
```

It looks like something has been fixed between 2.11.1 and 2.17.1. Looking at the [release notes](https://github.com/google/ExoPlayer/blob/release-v2/RELEASENOTES.md#2118-2020-08-25), it seems likely this was fixed in 2.11.8:
> * Fix `saiz` and `senc` sample count checks, resolving a "length mismatch" `ParserException` when playing certain protected FMP4 streams (https://github.com/google/ExoPlayer/issues/7592).

I would recommend upgrading to at least 2.11.8 and trying again.

**Comment by icbaker** [comment]
Duplicate of #7592
